app/upgrades/v17/upgrades.go

```diff
@@ -0,0 +1,327 @@
+		"umee-1":       sdkmath.ZeroInt(),  // TVL: ~150k |   <2.5M  | No rate limit
+	}
+
+	// Osmo transfer channel is required for new rate limits
+	OsmosisTransferChannelId = "channel-5"
+
+	// Constants for Prop 225
+	CommunityPoolGrowthAddress = "stride1lj0m72d70qerts9ksrsphy9nmsd4h0s88ll9gfphmhemh8ewet5qj44jc9"
+	LiquidityReceiver          = "stride1auhjs4zgp3ahvrpkspf088r2psz7wpyrypcnal"
+	Prop225TransferAmount      = sdk.NewInt(31_572_300_000)
+	Ustrd                      = "ustrd"
+)
+
+// CreateUpgradeHandler creates an SDK upgrade handler for v17
+func CreateUpgradeHandler(
+	mm *module.Manager,
+	configurator module.Configurator,
+	bankKeeper bankkeeper.Keeper,
+	distributionkeeper distributionkeeper.Keeper,
+	icqKeeper icqkeeper.Keeper,
+	ratelimitKeeper ratelimitkeeper.Keeper,
+	stakeibcKeeper stakeibckeeper.Keeper,
+) upgradetypes.UpgradeHandler {
+	return func(ctx sdk.Context, _ upgradetypes.Plan, vm module.VersionMap) (module.VersionMap, error) {
+		ctx.Logger().Info("Starting upgrade v17...")
+
+		ctx.Logger().Info("Migrating stakeibc params...")
+		MigrateStakeibcParams(ctx, stakeibcKeeper)
+
+		ctx.Logger().Info("Migrating host zones...")
+		if err := RegisterCommunityPoolAddresses(ctx, stakeibcKeeper); err != nil {
+			return vm, errorsmod.Wrapf(err, "unable to register community pool addresses on host zones")
+		}
+
+		ctx.Logger().Info("Deleting all pending slash queries...")
+		DeleteAllStaleQueries(ctx, icqKeeper)
+
+		ctx.Logger().Info("Reseting slash query in progress...")
+		ResetSlashQueryInProgress(ctx, stakeibcKeeper)
+
+		ctx.Logger().Info("Updating community pool tax...")
+		if err := IncreaseCommunityPoolTax(ctx, distributionkeeper); err != nil {
+			return vm, errorsmod.Wrapf(err, "unable to increase community pool tax")
+		}
+
+		ctx.Logger().Info("Updating redemption rate bounds...")
+		UpdateRedemptionRateBounds(ctx, stakeibcKeeper)
+
+		ctx.Logger().Info("Update rate limits thresholds...")
+		UpdateRateLimitThresholds(ctx, stakeibcKeeper, ratelimitKeeper)
+
+		ctx.Logger().Info("Adding rate limits to Osmosis...")
+		if err := AddRateLimitToOsmosis(ctx, ratelimitKeeper); err != nil {
+			return vm, errorsmod.Wrapf(err, "unable to add rate limits to Osmosis")
+		}
+
+		ctx.Logger().Info("Executing Prop 225, SHD Liquidity")
+		if err := ExecuteProp225(ctx, bankKeeper); err != nil {
+			return vm, errorsmod.Wrapf(err, "unable to execute prop 225")
+		}
+
+		return mm.RunMigrations(ctx, configurator, vm)
+	}
+}
+
+// Migrate the stakeibc params to add the ValidatorWeightCap parameter
+//
+// NOTE: If a parameter is added, the old params cannot be unmarshalled
+// to the new schema. To get around this, we have to set each parameter explicitly
+// Considering all mainnet stakeibc params are set to the default, we can just use that
+func MigrateStakeibcParams(ctx sdk.Context, k stakeibckeeper.Keeper) {
+	params := stakeibctypes.DefaultParams()
```

Comment:
The method of setting params (resetting everything to defaults) always makes me uncomfortable.

Have we verified all params are defaults? Is there any other way to do this?

Reply:
I've verified all parameters are default and also verified with the sdk team and osmosis devs that this is the only way

Reply:
Approach makes sense - will double check all the params in the morning

Reply:
double checked by @shellvish, marking this resolved